src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -151,36 +156,44 @@ private Task OnProjectFactoryCompletedAsync()
             AggregateWorkspaceProjectContext previousContextToDispose = null;
             using (await _gate.DisposableWaitAsync().ConfigureAwait(false))
             {
+                string newTargetFramework = null;
+                var projectProperties = await _commonServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
+
                 // Check if we have already computed the project context.
                 if (_currentAggregateProjectContext != null)
                 {
-                    // For non-cross targeting projects, we can use the current project context.
-                    // For cross-targeting projects, we need to verify that the TargetFrameworks for the current project context matches latest TargetFrameworks project property value.
+                    // For non-cross targeting projects, we can use the current project context if the TargetFramework hasn't changed.
+                    // For cross-targeting projects, we need to verify that the current project context matches latest frameworks targeted by the project.
                     // If not, we create a new one and dispose the current one.
 
                     if (!_currentAggregateProjectContext.IsCrossTargeting)
                     {
-                        return _currentAggregateProjectContext;
-                    }
-
-                    var projectProperties = await _commonServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
-                    var targetFrameworks = (string)await projectProperties.TargetFrameworks.GetValueAsync().ConfigureAwait(false);
+                        newTargetFramework = (string)await projectProperties.TargetFramework.GetValueAsync().ConfigureAwait(false);
```

Comment:
>TargetFramework [](start = 77, length = 15)

This may be a silly question but why do we only check for TargetFramework and not TargetFrameworks + TargetFrameworks here?